packages/docregistry/src/context.ts

```diff
@@ -58,20 +53,24 @@ export class Context<
     this._lastModifiedCheckMargin = options.lastModifiedCheckMargin || 500;
     const localPath = this._manager.contents.localPath(this._path);
     const lang = this._factory.preferredLanguage(PathExt.basename(localPath));
-    this._model = this._factory.createNew(
-      lang,
-      PageConfig.getOption('collaborative') === 'true'
-    );
-    const docProviderFactory = options.docProviderFactory;
-    this._provider = docProviderFactory
-      ? docProviderFactory({
-          path: this._path,
-          contentType: this._factory.contentType,
-          format: this._factory.fileFormat!,
-          model: this._model.sharedModel,
-          collaborative: this._model.collaborative
-        })
-      : new ProviderMock();
+
+    // Get shared model from IDrive.sharedModelFactory
+    const collaborative = PageConfig.getOption('collaborative') === 'true';
+    if (collaborative && this._factory.collaborative) {
+      const sharedFactory = this._manager.contents.getSharedModelFactory(
+        this._path
+      );
+      const sharedModel = sharedFactory?.createNew({
+        path: localPath,
+        format: this._factory.fileFormat!,
```

Comment:
Let's acknowledge it can be null

```suggestion
        format: this._factory.fileFormat,
```

Reply:
It can be `null` to support: https://github.com/jupyterlab/jupyterlab/pull/7596 - this means the API in `@jupyter/ydoc` needs to be updated. Make sure to comment what is the purpose of null.

We could also assume that `null` means collaboration is not possible and not calling `sharedFactory`. But this may hit us latter. So let's open the API and not support it for `YDrive`.

Reply:
> We could also assume that null means collaboration is not possible and not calling sharedFactory

We should because we need the format for the document room. We use the contentType, format, and path as id to identify the room every client connects for a shared document.

If we don't know the format, we can not connect to the corresponding room.

Reply:
In our specific implementation, I agree. But here, I only speak from an API point of view. Some developers may be interested into getting the `null` value because it needs that for providing a special shared model.